library/core/src/iter/traits/iterator.rs

```diff
@@ -1535,34 +1535,32 @@ pub trait Iterator {
     ///
     /// # Examples
     ///
-    /// Basic usage:
+    /// This demonstrates a use case that needs `by_ref`:
     ///
-    /// ```
-    /// let a = [1, 2, 3];
-    ///
-    /// let iter = a.iter();
-    ///
-    /// let sum: i32 = iter.take(5).fold(0, |acc, i| acc + i);
+    /// ```compile_fail,E0382
+    /// let a = [1, 2, 3, 4, 5];
+    /// let mut iter = a.iter();
+    /// let sum: i32 = iter.take(3).fold(0, |acc, i| acc + i);
```

Comment:
I previously had space between these lines in order to follow the "code in paragraphs" style; this feels really compact to me. It's also not entirely clear *why* we are adding more elements to the array and taking two more here.

Reply:
> It's also not entirely clear why we are adding more elements to the array and taking two more here.

I think it's because it's showing that we want to consume only part of the iterator here and then consume more further down.

Reply:
In terms of the grouping, maybe it would be better as

```rust
let a = [1, 2, 3, 4, 5];
let mut iter = a.iter();

let sum: i32 = iter.take(3).fold(0, |acc, i| acc + i);
assert_eq!(sum, 6);
```

Reply:
Sure, that seems fine.

Reply:
Changed the grouping.